server.js

```diff
@@ -0,0 +1,27 @@
+var express = require('express'),
+    morgan = require("morgan"),
+    bodyParser = require('body-parser');
+
+var app = express().disable("x-powered-by");
+
+if (process.env.NODE_ENV !== "production") {
+  app.use(morgan("dev"));
+}
+
+app.use(bodyParser.json());
+
+app.get('/', function(req, res) {
+  res.send('pong');
+});
+
+app.post('/metadata', function(req, res){
+  console.log(req.body);
+  res.setHeader('Content-Type', 'text/plain');
+  res.write('you posted:\n');
+  res.end(JSON.stringify(req.body, null, 2));
```

Comment:
Explicit returns are good (especially within callbacks).

In this case:

```javascript
return res.end(JSON.stringify(req.body, null, 2));
```

There's actually shorthand here for JSON responses (without the "you posted" bit):

```javascript
  console.log(req.body);
  return res.json(req.body);
```

(For testing APIs that return JSON, I use [jq](http://stedolan.github.io/jq/) all the time for pretty printing and syntax highlighting: `curl http://whatever/something.json | jq .`)